Content.Client/UserInterface/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,441 @@
+        };
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    private void PlaySizeAnimation(Control button, Vector2 size, string playKey, string? stopKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(_focusAniTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.ButtonSize),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(button.Size, 0f),
+                        new AnimationTrackProperty.KeyFrame(size, _focusAniTime)
+                    }
+                }
+            }
+        };
+
+        if (stopKey != null && button.HasRunningAnimation(stopKey))
+            button.StopAnimation(stopKey);
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        base.Draw(handle);
+
+        foreach (var child in Layout.Children)
+        {
+            var button = (RadialItem)child;
+            LayoutContainer.SetPosition(child, button.Offset - (button.Size/2));
+        }
+        LayoutContainer.SetPosition(CloseButton, CloseButton.Offset - (CloseButton.Size/2));
+        LayoutContainer.SetPosition(ActionLabel,
+            new Vector2(0 - (GetTextWidth(ActionLabel) / 4), GetDistance(4.5f) ));
+    }
+
+    protected override void FrameUpdate(FrameEventArgs args)
+    {
+        if (!_isAttached)
+            return;
+
+        base.FrameUpdate(args);
+
+        var entityManager = IoCManager.Resolve<IEntityManager>();
+        var eyeManager = IoCManager.Resolve<IEyeManager>();
+
+        if (entityManager.Deleted(_attachedEntity))
+        {
+            Timer.Spawn(0, Die);
+            return;
+        }
+
+        if (!entityManager.TryGetComponent<TransformComponent>(_attachedEntity, out var xform) || xform.MapID != eyeManager.CurrentMap)
+        {
+            return;
+        }
+
+        var localPlayer = IoCManager.Resolve<IPlayerManager>().LocalPlayer;
+        if (localPlayer == null)
+            return;
+        // Check distance beetween entities
+        if (entityManager.TryGetComponent<TransformComponent>(localPlayer.ControlledEntity, out var myxform))
+        {
+            var onePoint = xform.WorldPosition;
+            var twoPoint = myxform.WorldPosition;
+            var distance = (onePoint - twoPoint).Length();
+
+            if (DistanceAvaible < distance)
+            {
+                Timer.Spawn(0, Die);
+                return;
+            }
+        }
+
+        var offset = (-eyeManager.CurrentEye.Rotation).ToWorldVec() * -VerticalOffset;
+        var worldPos = xform.WorldPosition + offset;
```

Comment:
This is going to break with multi-viewports, plus radial itself shouldn't care about anything and a separate system or UI controller should be managing it, not the control itself.